[
    {
        "content": "<p>Recently, I made too many changes to <code>topology.order</code> while porting. First, <strong>don't make this mistake</strong>. Second, here is how I recovered from it.</p>\n<ul>\n<li>Backport changes to <code>mathlib</code>, wait for the PR to be merged and <code>mathlib3port</code> to be regenerated (I've slightly sped up the process by manually starting the actions <a href=\"https://github.com/leanprover-community/mathport/actions/workflows/predata.yml\">1</a> <a href=\"https://github.com/leanprover-community/mathport/actions/workflows/build.yml\">2</a>).</li>\n<li>Merge <code>master</code> into <code>port/*</code>.</li>\n<li>Edit <code>start_port.sh</code> to remove the safety check about porting being already started and use another branch name (<code>port/tmp/*</code> in my case).</li>\n<li>Run the modified script.</li>\n<li>Checkout <code>port/*</code></li>\n<li>Run <code>git reset --mixed port/tmp/branch</code></li>\n<li>Revert modification to the line with SHA sum.</li>\n<li>Review the diff and accept/reject modifications one by one (I use Emacs <code>magit</code>, probably VSCode has similar functionality).</li>\n<li>Forse-push to Github, delete temporary branch.</li>\n</ul>",
        "id": 324896844,
        "sender_full_name": "Yury G. Kudryashov",
        "timestamp": 1675167666
    },
    {
        "content": "<p>How about we ban making this mistake?</p>",
        "id": 324937157,
        "sender_full_name": "Kevin Buzzard",
        "timestamp": 1675177805
    }
]